martin/src/utils/utilities.rs

```diff
@@ -28,3 +32,39 @@ pub fn encode_brotli(data: &[u8]) -> Result<Vec<u8>, std::io::Error> {
     encoder.write_all(data)?;
     Ok(encoder.into_inner())
 }
+
+pub fn parse_base_path(path: &String) -> MartinResult<String> {
```

Comment:
several things:
* never pass `&String` as a parameter - use `&str` instead
* you don't need to clone anything here, except when returning an error
* use `.trim_end_matches('/')` to convert one `&str` into another `&str`